contracts/MemeFactory.sol

```diff
@@ -201,11 +208,60 @@ abstract contract MemeFactory {
         (positionId, liquidity, , ) = IUniswapV3(uniV3PositionManager).mint(params);
     }
 
+    function _getTwapFromOracle(address pool) internal view returns (uint256 priceX96) {
+        // Query the pool for the current and historical tick
+        uint32[] memory secondsAgos = new uint32[](2);
+        // Start of the period
+        secondsAgos[0] = SECONDS_AGO;
+
+        // Fetch the tick cumulative values from the pool
+        (int56[] memory tickCumulatives, ) = IUniswapV3(pool).observe(secondsAgos);
+
+        // Calculate the average tick over the time period
+        int56 tickCumulativeDelta = tickCumulatives[1] - tickCumulatives[0];
+        int24 averageTick = int24(tickCumulativeDelta / int56(int32(SECONDS_AGO)));
```

Comment:
different ints bottom and top, intended?

Reply:
Why? from line 221 the top is `int56` as well. the result is then brought to `int24`.

Reply:
ok, sorry didn't see brackets correctly